packages/popover2/src/tooltip2.tsx

```diff
@@ -87,7 +88,24 @@ export class Tooltip2<T> extends React.PureComponent<Tooltip2Props<T>> {
     private popover: Popover2<T> | null = null;
 
     public render() {
-        const { children, intent, popoverClassName, ...restProps } = this.props;
+        // if we have an ancestor Tooltip2Context, we should take its state into account in this render path,
+        // it was likely created by a parent ContextMenu2
+        return (
+            <Tooltip2Context.Consumer>
+                {([state]) => <Tooltip2Provider initialState={state}>{this.renderPopover}</Tooltip2Provider>}
```

Comment:
what happens if you have nested tooltips? (acknowledging that it's not the best UX, but not ruling out that its possibility 😬 )

Reply:
I don't think it would be a possibility in terms of tooltips nested inside tooltip _content_, but nesting of tooltip targets could be possible (just like nesting of ContextMenu2 targets, as demonstrated in the ContextMenu2 docs example). I'll add a test for it

Reply:
oops yes, i meant targets, thanks for clarifying!